### ISSUE TITLE
Include content by document structure

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#1328]
It would be useful to be able to include content from another AsciiDoc file by structural boundaries rather than lines or tagged regions. This would be a very simple way to cherry pick a section or block from another document and have it included into the current document. You wouldn't have to worry about where the content is within the file so long as you can reference it by its id.

Assume we have content in another file that looks like:

.example.adoc
```
[#sec-a]
== Section A

content

[#sec-b]
== Section B

content

[#sec-c]
== Section C

content
```

We could include the content of Section B using:

```
include::example.adoc[refid=sec-b]
```

Or perhaps we could use the syntax:

```
include::sec-b@example.adoc[]
```

(I kind of like the second option better).

You can grab arbitrary content by wrapping the content in an open block (or any block for that matter) and assigning it an id to reference.

I suppose we could allow for only the child elements to be extracted as well using something like:

```
include::sec-b::content@example.adoc[]
```

**Comment by lordofthejars** [comment]
This is a really good idea. I would say that event it should works without being mandatory to add the [#sec-b] to title. Something like cross reference that you can use the id you have given or the title of the chapter.

Of course the problem with the title of the chapter is that it can be white spaces, so then the first option I think it would be the desired one.

**Comment by mojavelinux** [comment]
> it should works without being mandatory to add the [#sec-b] to title

Yes, indeed. I just included those to make the example more clear. Implicit ids will work as well.

**Comment by mojavelinux** [comment]
If nothing else, we need this for section content.
